### PR TITLE
added readme and test report and restructured some code

### PR DESCRIPTION
**Pull request by kpunit020**

### CODE REVIEW

File: Material_FAB/index.ets [hunk 1]
```diff
@@ -1,3 +1,20 @@
+/*
+ * Copyright (C) 2020-21 Application Library Engineering Group
```
Comment (samuel-abhishek) [inline Material_FAB/index.ets]: Modify the copyright to 2021-2022. Applicable to all the files.

File: README.md [hunk 2]
```diff
@@ -1,38 +1,65 @@
...
+```ets
+import { FAB_regular, FAB_extended }  from '@ohos/Material_FAB'
+```
 
-## <FAB_regular>
 
-Implementation of Regular FAB buttons with size options as *small*, *medium* and *large*.
 ## Screenshots
 
+![img.png](images/sample_small_fab.png)   ![img.png](images/sample_medium_fab.png)
```
Comment (samuel-abhishek) [inline README.md]: Include a gif to show the ripple effect implemented on the FAB during onClick

File: README.md [hunk 3]
```diff
@@ -1,38 +1,65 @@
...
-| `onClick` | `callback` | Give behaviour for the click event |
+| `fabSize` | `string` | specifies the type of FAB_regular from small, medium and large. Defaults to *medium*.  |
+| `position` | `Position` | Specify the position of the FAB. Defaults to [0, 0]. |
+| `backgroundColor` | `ResourceColor` | Specify the background color for the FAB. Defaults to *Blue*. |
+| `border` | `BorderOptions` | give custom border to the FAB. |
+| `fabCustomSize` | `SizeOptions` | Give custom size for the FAB. |
+| `margin` | `Margin` | Give margin to the FAB. |
 
 
+#### Other properties
```
Comment (samuel-abhishek) [inline README.md]: Remove the test report from the project repo. It will be added as an artifact during github release anyways.

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Ok. Check the comments and do the required changes

File: Material_FAB/src/main/ets/components/MainPage/FAB_extended.ets [hunk 4]
```diff
@@ -18,38 +18,57 @@ import { rippleUtils } from './ripple'
...
-  onClick: (event?: ClickEvent) => void = (event) => { // callback for onClick
-    console.log('FAB_extended clicked')
-  }
 
+
+  /*
+  set values for all the properties
+   */
   aboutToAppear() {
+    this.iconPadding = this.iconPadding? this.iconPadding : {left : 20 , right : -8}
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/FAB_extended.ets]: Mark all dimentions as VP

File: Material_FAB/src/main/ets/components/MainPage/FAB_extended.ets [hunk 5]
```diff
@@ -18,38 +18,57 @@ import { rippleUtils } from './ripple'
...
+    this.iconSize = this.iconSize? this.iconSize : { width: 24, height: 24 }
+
+    this.fontColor = this.fontColor? this.fontColor : Color.White
+    this.fontSize = this.fontSize? this.fontSize : 24
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/FAB_extended.ets]: Format all the code.

File: Material_FAB/src/main/ets/components/MainPage/FAB_regular.ets [hunk 6]
```diff
@@ -61,7 +59,21 @@ export struct FAB_regular {
     },
   }
 
+  private check_fabSize_type() {
+    if(this.fabSize != 'small' && this.fabSize != 'medium' && this.fabSize != 'large')
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/FAB_regular.ets]: Suggest to define as enum and use them for better readebility.
Refer to,
https://github.com/Applib-OpenHarmony/Material_Ui_Checkbox/pull/1/files#diff-c999b5fc24fe70d53d7a39c368b3fed81cb13cd7dff1427415c224348f9cd4f0

File: README.md [hunk 7]
```diff
@@ -52,3 +79,142 @@ struct small_FAB_Demo {
...
+```
+import {FAB_regular} from '@ohos/Material_FAB'
+
+@Entry
+@Component
+struct large_FAB_Demo {
+  build() {
+    Column(){
+      FAB_regular({
+        fabIcon : $r('app.media.add_icon'),
```
Comment (archanaatr) [inline README.md]: Suggest to create as model for user inputs in the custom component.
Reference: https://github.com/Applib-OpenHarmony/Material_Ui_Checkbox/pull/1/files#diff-8ec4167e79402b1c41a8952ee93b5a535596d04dab9e2befccf8c3ab5c61b6dd

**Review by archanaatr — COMMENTED** [review]
please check the comments.

**Review by samuel-abhishek — APPROVED** [review]
LGTM

**Review by samuel-abhishek — APPROVED** [review]
okay

File: Material_FAB/index.ets [hunk 8]
```diff
@@ -1,3 +1,17 @@
-export { rippleUtils } from './src/main/ets/components/MainPage/ripple'
```
Comment (archanaatr) [inline Material_FAB/index.ets:1]: Change the lib folder name from Material_FAB to material-fab

File: Material_FAB/index.ets [hunk 9]
```diff
@@ -1,3 +1,17 @@
...
-export {FAB_regular} from './src/main/ets/components/MainPage/FAB_regular'
-export {FAB_extended} from './src/main/ets/components/MainPage/FAB_extended'
```
Comment (archanaatr) [inline Material_FAB/index.ets:3]: package.json update the library name as per the format shared

File: Material_FAB/src/main/ets/components/MainPage/fabUtils/fabTypes.ets [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum FabType {
+  small,
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/fabUtils/fabTypes.ets]: suggest to define all enum values in caps

File: Material_FAB/src/main/ets/components/MainPage/ripple/rippleModel.ets [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+    getRippleColor(): ResourceColor {
+      return this.rippleColor
+    }
+
+    setRippleColor(rippleColor : ResourceColor) : Model {
+      this.rippleColor = (rippleColor != null && rippleColor != undefined)? rippleColor : DEFAULT_RIPPLE_COLOR
+      return this
+    }
+
+    getCircleSize(): number{
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/ripple/rippleModel.ets]: format all the ETS files. (Open each ets file > Ctrl + Alt + L)

File: Material_FAB/src/main/ets/components/MainPage/ripple/ripple.ets [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+import { RippleModel } from './rippleModel'
+
+@Component
+export struct Ripple {
+  @State model: RippleModel.Model = undefined
+  @Link private show_ripple : boolean
+
+  private show_animation() {
+    animateTo({
+      duration: this.model.getRippleDuration(),
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/ripple/ripple.ets:25]: better to add validation on model object, recheck for required validations in other files as well.

File: Material_FAB/index.ets (hunk 8 above)
Comment (kpunit020) [reply Material_FAB/index.ets:1]: library name can't have '-' in it

File: Material_FAB/src/main/ets/components/MainPage/fabUtils/fab_extended_model.ets [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+import { CommonProperties } from './fabCommonProperties'
+
+namespace FabExtendedModel {
+  const DEFAULT_FAB_LABEL : string | Resource = 'Add'
+  const DEFAULT_FONT_COLOR : ResourceColor = Color.White
+  const DEFAULT_FONT_STYLE : FontStyle = FontStyle.Normal
+  const DEFAULT_FONT_SIZE : number | string | Resource = '24vp'
+  const DEFAULT_FONT_WEIGHT : FontWeight = FontWeight.Bold
+  const DEFAULT_TEXT_PADDING : Padding = {left : '20vp', right : '20vp'}
+  const DEFAULT_FAB_MAX_WIDTH : number = 200
```
Comment (archanaatr) [inline Material_FAB/src/main/ets/components/MainPage/fabUtils/fab_extended_model.ets]: To be marked as vp?

**Review by archanaatr — COMMENTED** [review]
Please check the comments.

File: Material_FAB/index.ets (hunk 9 above)
Comment (kpunit020) [reply Material_FAB/index.ets:3]: these are the previous changes

File: material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+        show_ripple: $show_ripple
+      })
+
+    }
+    .constraintSize({ minWidth: '80vp', maxWidth: this.model.getFabMaxWidth() })
+    .height('56vp')
+    .backgroundColor(this.model.getBackgroundColor())
+    .border(this.model.getBorder())
+    .zIndex(Number.MAX_VALUE)
+    .shadow({ radius: 30, color: Color.Gray, offsetY: 16 })
```
Comment (archanaatr) [inline material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets]: Mark all dimentions as vp and font sizes as fp

File: material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+@Component
+export struct FAB_extended {
+  @State model: FabExtendedModel.Model = undefined
+  @State show_ripple: boolean = false
+  onClick: (event?: ClickEvent) => void
+
+  build() {
+    Row() {
+      if (this.model.getFabIcon()) {
```
Comment (archanaatr) [inline material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets]: Add validation on model to check if its not null and not undefined... add similar validation for all required variables in all files

File: material-fab/src/main/ets/components/MainPage/fabUtils/fabCommonProperties.ets [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RippleModel } from '../ripple/rippleModel'
+
+const DEFAULT_FAB_ICON: ResourceStr = undefined
```
Comment (archanaatr) [inline material-fab/src/main/ets/components/MainPage/fabUtils/fabCommonProperties.ets:18]: check for validations at their usage

File: material-fab/src/main/ets/components/MainPage/ripple/rippleModel.ets [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace RippleModel {
+  const DEFAULT_RIPPLE_DURATION: number = 300
+  const DEFAULT_IS_BOUNDED: boolean = true
+  const DEFAULT_RIPPLE_COLOR: ResourceColor = Color.White
+  const DEFAULT_CIRCLE_SIZE: number = 10
```
Comment (archanaatr) [inline material-fab/src/main/ets/components/MainPage/ripple/rippleModel.ets]: DEFAULT_CIRCLE_SIZE to be marked as vp?

**Review by archanaatr — COMMENTED** [review]
please check the comments.

File: material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets (hunk 14 above)
Comment (kpunit020) [reply material-fab/src/main/ets/components/MainPage/fabExtendedComponent.ets]: can't give shadow properties as vp

File: material-fab/src/main/ets/components/MainPage/fabUtils/fabCommonProperties.ets (hunk 16 above)
Comment (kpunit020) [reply material-fab/src/main/ets/components/MainPage/fabUtils/fabCommonProperties.ets:18]: changed value of default variables from undefined to some other value
only DEFAULT_FAB_ICON is undefined for now and validations have been added at it's usage

**Pull request merged**